lib/Github/HttpClient/CachedHttpClient.php

```diff
@@ -105,4 +114,31 @@ public function getLastResponse($force = false)
 
         return ($force) ? $lastResponse : $this->lastCachedResponse;
     }
+
+    /**
+     * Format the path and add query parameters if they exist.
+     *
+     * @param string $path
+     * @param array  $options
+     * @return void
+     */
+    private function formatPath($path, array $options)
+    {
+        $this->path = $path;
+
+        if (array_key_exists('query', $options) && !empty($options['query'])) {
+            $this->path .= '?';
+
+            $i = 0;
+            foreach ($options['query'] as $key => $value) {
```

Comment:
Look, https://developer.github.com/v3/repos/commits/ the list of available parameters for the commits endpoint.
You say it looks like a bug but it is perfectly working according to Github specs.

So you suggest the following?:

- If no query parameters inside `$option['query']` urlencode `$path` (because `$path` can still contain query parameters). How do you manager if a user like don't encode by hand the parameters?
- If query parameters inside `$option['query'] urlencode `$path`

Can you please make an effort in your answers so we can go forward asap with this pull request :)